### PR TITLE
Add note about having to run `astro build` before `astro preview`

### PR DESCRIPTION
**Pull request by hirasso**
#### Description (required)

Yesterday [I got bitten](https://github.com/withastro/astro/issues/9297) by not having completely understood that `astro preview` doesn't automatically `build` the project. This PR adds a `caution` box to save others some time:

CleanShot 2023-12-05 at 10 02 52@2x

#### Related issues & labels (optional)

- https://github.com/withastro/astro/issues/9297
- Suggested label: add new content

### CODE REVIEW

**Review by sarah11918 — APPROVED** [review]
Thank you for this contribution, @hirasso ! We appreciate you taking the time to add some helpful content for the next person! 🙌 

I have edited the format of your content to be more in line with the rest of our docs. We rarely add notes, which are for tangential information, and almost never add cautions which are for truly alarming/security issues to watch out for. 

We prefer to incorporate important content into the main text itself, instead of new content being "oh, and another thing!" or "Oh, and note this!" This makes for a better reading experience, and if it's important content, it should be said right the first time.

I've reviewed our documentation both on `astro build` and `astro preview` and added a bit of extra context right in the description to show that `preview` is meant to be used in conjunction with `build`. Rather than a scary warning message, I have linked the two together.

I do think this is helpful extra context to have, and we are happy you contributed it! Welcome to Team Docs! 🥳

**Comment by hirasso** [comment]
As an aside note: very interesting to read your thoughts about when to use a note/caution. I just plain love the Astro docs! ❤️

File: src/content/docs/en/reference/cli-reference.mdx [hunk 1]
```diff
@@ -224,9 +224,9 @@ Can be combined with the [common flags](#common-flags) documented below.
 
 ## `astro preview`
 
-Starts a local server to serve your static `dist/` directory.
+Starts a local server to serve the contents of your static `dist/` directory.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:227]:
```suggestion
Starts a local server to serve the contents of your static directory (`dist/` by default).
```

File: src/content/docs/en/reference/cli-reference.mdx [hunk 2]
```diff
@@ -224,9 +224,9 @@ Can be combined with the [common flags](#common-flags) documented below.
...
 
-This command is useful for previewing your build locally, before deploying it. It is not designed to be run in production. For help with production hosting, check out our guide on [Deploying an Astro Website](/en/guides/deploy/).
+This command is useful for previewing your [build](#astro-build) locally, before deploying it. It is not designed to be run in production. For help with production hosting, check out our guide on [Deploying an Astro Website](/en/guides/deploy/).
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:229]:
```suggestion
This command allows you to preview your site locally [after building](#astro-build) to catch any errors in your build output before deploying it. It is not designed to be run in production. For help with production hosting, check out our guide on [Deploying an Astro Website](/en/guides/deploy/).
```

**Comment by sarah11918** [comment]
Hi @hirasso, See what you think now! I tried to add a little bit more context in both the opening line and the next paragraph.

**Comment by hirasso** [comment]
@sarah11918 now it's much more clear! Ready to be merged I'd say 😊

**Pull request merged**